utils/gddrescue/Makefile

```diff
@@ -0,0 +1,46 @@
+#
+# Copyright (C) 2018 Lucian Cristian
+#
+
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=gddrescue
+PKG_VERSION:=1.22
+PKG_RELEASE:=1
+
+PKG_SOURCE:=$(PKG_NAME)_$(PKG_VERSION).orig.tar.xz
+PKG_SOURCE_URL:=http://http.debian.net/debian/pool/main/g/$(PKG_NAME)
+PKG_HASH:=a715488f77ae201099ff99360baf7d909bce0fa6c43588e076e830fec93d09f1
+PKG_MAINTAINER:=Lucian Cristian <lucian.cristian@gmail.com>
+
+PKG_LICENSE:=GPL-2.0
+PKG_LICENSE_FILES:=LICENSE
+
+PKG_INSTALL:=1
+
+include $(INCLUDE_DIR)/package.mk
+
+define Package/gddrescue
+  SECTION:=utils
+  CATEGORY:=Utilities
+  TITLE:=Data recovery tool
+  URL:=https://www.gnu.org/software/ddrescue/
+  DEPENDS:= +libstdcpp
+endef
+
+define Package/gddrescue/description
+	GNU ddrescue is a data recovery tool. It copies data from one file
+	or block device (hard disc, cdrom, etc) to another, trying to rescue
+	the good parts first in case of read errors.
+endef
+
+MAKE_FLAGS += \
```

Comment:
More needs to be done here. From brief compilation I'm getting:

```
g++-uc  -Wall -W -O2 -c -o arg_parser.o arg_parser.cc
```

none of the CFLAGS are getting passed.

Reply:
added something like this

```
define Build/Compile
        $(MAKE) -C $(PKG_BUILD_DIR) \
                CC="$(TARGET_CXX)" \
                CFLAGS="$(TARGET_CFLAGS) $(TARGET_CPPFLAGS) -fno-rtti" \
                all
endef
```
but then still
```
Package gddrescue is missing dependencies for the following libraries:
libc.so.6
libstdc++.so.6

```

Reply:
I'm getting this when trying to compile with uclibc++:

```
creating config.status
creating Makefile
VPATH = .
prefix = /usr
exec_prefix = /usr
bindir = /usr/bin
datarootdir = $(prefix)/share
infodir = /usr/info
mandir = /usr/man
CXX = g++
CPPFLAGS = 
CXXFLAGS = -Wall -W -O2
LDFLAGS = 
OK. Now you can run make.
```

CXX is then later changed to gcc+-uc by the build system but not CXX or CPPFLAGS. Or LDFLAGS.

Also, it turns out there's a configure script you can modify.

Reply:
the values are hardcoded in configure, adding
```
CXXFLAGS = -Wall -W -O2 -flto -fno-rtti
LDFLAGS = -Wl,--as-needed -flto
```
still results in
```
Package gddrescue is missing dependencies for the following libraries:
libc.so.6
libstdc++.so.6

```

Reply:
Here's the patch that I use:

https://gist.github.com/neheb/eacb3dbec32357f2be9cec3b1ceeda9c

It works properly. That is, it fails compilation. Until the uclibc++ issue can be figured out, we would need to stick with libstdcpp.

Reply:
The configure script also has an interesting --enable-non-posix option which could be useful.

Reply:
I asked upstream to see what they think: http://lists.gnu.org/archive/html/bug-ddrescue/2019-03/msg00000.html

Reply:
ahh, it used g++
setting 

```
CXX = g++-uc
CXXFLAGS = -Wall -W -Os -flto -fno-rtti
LDFLAGS = -flto -nodefaultlibs
```

still errors with `/home/md10/build/sunxi/staging_dir/target-arm_cortex-a7+neon-vfpv4_musl_eabi/usr/include/uClibc++/vector:270:6: error: no matching function for call to 'Sblock::Sblock()'`